src/recorder/jamrecorder.cpp

```diff
@@ -396,14 +396,30 @@ void CJamRecorder::Start() {
     // Ensure any previous cleaning up has been done.
     OnEnd();
 
+    QString error;
+
     // needs to be after OnEnd() as that also locks
     ChIdMutex.lock();
     {
-        currentSession = new CJamSession( recordBaseDir );
-        isRecording = true;
+        try
+        {
+            currentSession = new CJamSession( recordBaseDir );
+            isRecording = true;
+        }
+        catch ( const CGenErr& err )
+        {
+            currentSession = nullptr;
+            error = err.GetErrorText();
+        }
     }
     ChIdMutex.unlock();
 
+    if ( !currentSession )
+    {
+        emit RecordingFailed ( error );
```

Comment:
Ah, sorry...  If this is _only_ emitted on start of recording, I'd like it called `RecordingFailedToStart`.  That way, if ever there's a failure during recording, it can be named - i.e. signalled - separately (as handling _may_ need to be different).

Reply:
Only worry about this if the change to the `#include "../server.h"` is relevant.